polkadot/node/core/runtime-api/src/cache.rs

```diff
@@ -446,6 +448,14 @@ impl RequestResultCache {
 		self.minimum_backing_votes.insert(session_index, minimum_backing_votes);
 	}
 
+	pub(crate) fn client_features(&mut self, key: &Hash) -> Option<vstaging::ClientFeatures> {
+		self.client_features.get(key).copied()
+	}
+
+	pub(crate) fn cache_client_features(&mut self, key: Hash, features: vstaging::ClientFeatures) {
```

Comment:
I don't think we need those to be able to change on a per block basis. We should make ClientFeatures session buffered.

Reply:
done. please have a look